models/Document/Tag/Pdf.php

```diff
@@ -70,8 +70,8 @@ public function getDataForResource()
     public function getDataEditmode()
     {
         $pages = 0;
-        if ($asset = Asset::getById($this->id)) {
-            /** @var Asset\Document $asset */
+
+        if (/** @var Asset\Document $asset */ $asset = Asset::getById($this->id)) {
```

Comment:
Why not put that comment above the conditional statement and not inside it? It looks very weird to have a comment in between code ...

Reply:
because Mr. Stan is not happy about that. maybe there is a better solution?

Reply:
So something like
```php
/** @var Asset\Document $asset */
if ($asset = Asset::getById($this->id)) {
    // ...
}
```
or
```php
$asset = Asset::getById($this->id);

if ($asset instanceof Asset) {
    // ...
}
```
doesn't work?

Reply:
Negative :-)

Reply:
@aarongerig The second approach should work
@weisswurstkanone We can use Asset\Document::getById() and change the return to static|null. Like in AbstractObject:getById(). So we don't need the comments, right?

Reply:
@aarongerig yes, second one works but this is something you sometimes don't want

Reply:
I'm sorry, of course I mean to write:

```php
$asset = Asset::getById($this->id);

if ($asset instanceof Asset\Document) {
    $pages = $asset->getPageCount();
}
```

Just tested that on my machine and it works. :)

Reply:
that's what I said. but imagine a if/else/... scenario which is a bit more complicated

Reply:
@blankse yep, this static stuff works. I would go for it.